crypto/bn/bn_print.c

```diff
@@ -32,14 +32,14 @@ char *BN_bn2hex(const BIGNUM *a)
     }
     p = buf;
     if (a->neg)
-        *(p++) = '-';
+        *p++ = '-';
     for (i = a->top - 1; i >= 0; i--) {
         for (j = BN_BITS2 - 8; j >= 0; j -= 8) {
             /* strip leading zeros */
-            v = ((int)(a->d[i] >> (long)j)) & 0xff;
-            if (z || (v != 0)) {
-                *(p++) = Hex[v >> 4];
-                *(p++) = Hex[v & 0x0f];
+            v = 0xff & (int)(a->d[i] >> (long)j);
```

Comment:
Well, if we are taking this path, I'd insist on removing `(long)`. But more importantly (as already discussed, and since we are turning these stones) unsigned long to int is not exactly desired, and from *formal* viewpoint we are better off with `(int)(a->d[i] >> j) & 0xff)`, i.e. mask first *then* cast, which guarantees that [masked] unsigned long can be represented in int. In such case omission of cast would be appropriate, *but* unfortunately *some* compilers might issue a warning, so explicit cast has to stay :-(